### PR TITLE
Bump default FreeBSD version to 12.x

### PR DESCRIPTION
**Pull request by ibuclaw**
FreeBSD 11 has been retired from the main pkg repositories (https://pkg.freebsd.org/).

### CODE REVIEW

**Review by kinke — APPROVED** [review]

**Comment by kinke** [comment]
Argh. So autotester apparently uses v11 and needs an explicit flag.

**Comment by kinke** [comment]
Oh and DMD v2.079 apparently doesn't work on FreeBSD 12, producing a crashing `build.d` executable. Maybe use GDC for the FreeBSD bootstrapping job instead?

**Comment by ljmf00** [comment]
> Oh and DMD v2.079 apparently doesn't work on FreeBSD 12, producing a crashing `build.d` executable. Maybe use GDC for the FreeBSD bootstrapping job instead?

Perhaps a problem on 12.3? We switched from 12.2 to 12.3 and only tested 12.3 when bootstrapping. Just a thought tho since it could be an old bug in the bootstrapping version.

File: .cirrus.yml [hunk 1]
```diff
@@ -135,32 +135,32 @@ macos10_task:
...
   freebsd_instance:
-    image_family: freebsd-12-2
+    image_family: freebsd-13-0
     cpu: 4
     memory: 8G
   timeout_in: 60m
   environment:
     OS_NAME: freebsd
-    CI_DFLAGS: -version=TARGET_FREEBSD12
+    CI_DFLAGS: -version=TARGET_FREEBSD13
```
Comment (kinke) [inline .cirrus.yml:146]: Where are these actually used? I grepped for `TARGET_FREEBSD` in dmd/druntime/Phobos, no relevant hits.

Comment (ljmf00) [reply .cirrus.yml:146]: I belive its just for future convenience, but yeah, probably worth to be on the build system itself rather than the CI config

Comment (ibuclaw) [reply .cirrus.yml:146]: Apparently someone removed the code that used this version condition.

Comment (ibuclaw) [reply .cirrus.yml:146]: This probably also means that the current 12.x pipeline is likely a sham, as the built compiler is default targetting FreeBSD 11.

Comment (kinke) [reply .cirrus.yml:146]: Uh yeah probably. - This probably fell under the carpet with `-target=…` cross-compilation support. `CI_DFLAGS` are most likely not sufficient in that case (only affecting dmd/druntime/Phobos builds, not the default target of the built DMD).

Comment (ljmf00) [reply .cirrus.yml:146]: That's correct. `git log --pickaxe-regex` tells me https://github.com/dlang/dmd/commit/ee446a03971c99a99de15b28930e205103063260 from #12508 .

Comment (ljmf00) [reply .cirrus.yml:146]: Cross referencing https://github.com/dlang/dmd/pull/14071

**Comment by ibuclaw** [comment]
We'll bump the image post-release.

**Pull request closed without merging**